app/models/solidus_afterpay/payment_method.rb

```diff
@@ -30,11 +31,19 @@ def try_void(payment)
     end
 
     def available_for_order?(order)
+      return false if order.line_items.any?{ |item| excluded_products.include? item.product.id }
```

Comment:
A performance optimization could be to get the product_id from the variant

```ruby
# Spree::LineItem#product.id

irb(main):005:0> Spree::Order.last.line_items.first.product.id
  Spree::Order Load (0.2ms)  SELECT "spree_orders".* FROM "spree_orders" ORDER BY "spree_orders"."id" DESC LIMIT ?  [["LIMIT", 1]]
  Spree::LineItem Load (0.2ms)  SELECT "spree_line_items".* FROM "spree_line_items" WHERE "spree_line_items"."order_id" = ? ORDER BY "spree_line_items"."created_at" ASC, "spree_line_items"."id" ASC LIMIT ?  [["order_id", 2], ["LIMIT", 1]]
  Spree::Product Load (0.1ms)  SELECT "spree_products".* FROM "spree_products" INNER JOIN "spree_variants" ON "spree_products"."id" = "spree_variants"."product_id" WHERE "spree_variants"."id" = ? LIMIT ?  [["id", 4], ["LIMIT", 1]]
```

VS

```ruby
# Spree::LineItem#variant.product_id

irb(main):004:0> Spree::Order.last.line_items.first.variant.product_id
  Spree::Order Load (0.2ms)  SELECT "spree_orders".* FROM "spree_orders" ORDER BY "spree_orders"."id" DESC LIMIT ?  [["LIMIT", 1]]
  Spree::LineItem Load (0.1ms)  SELECT "spree_line_items".* FROM "spree_line_items" WHERE "spree_line_items"."order_id" = ? ORDER BY "spree_line_items"."created_at" ASC, "spree_line_items"."id" ASC LIMIT ?  [["order_id", 2], ["LIMIT", 1]]
  Spree::Variant Load (0.2ms)  SELECT "spree_variants".* FROM "spree_variants" WHERE "spree_variants"."id" = ? LIMIT ?  [["id", 4], ["LIMIT", 1]]
```

as you can see there is one less `INNER JOIN`

Reply:
Wooow... Skills 👍